src/legacy/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -441,9 +444,16 @@ function VisEditor(
   }
 
   $scope.updateQueryAndFetch = function ({ query, dateRange }) {
-    timefilter.setTime(dateRange);
-    $state.query = query;
-    $scope.fetch();
+    const isUpdate = (
+      (query && !_.isEqual(query, $state.query)) ||
+      (dateRange && !_.isEqual(dateRange, $scope.timeRange))
+    );
+
+    if (query && !_.isEqual(query, $state.query)) $state.query = query;
```

Comment:
do we still need this checks here if w have the checks in embedded_visualize_handler ?

Reply:
We still need to check if nothing has changed in order to manually trigger a fetch, but you're right, we don't need to check before we set them. Fixed in cc10e25445.